### PR TITLE
Remove t_cose work-arounds for Notary style claims

### PR DESCRIPTION
**Pull request by darracott**
- [ISSUE-100](https://github.com/microsoft/scitt-ccf-ledger/issues/100): Update CCF from [3.0.2 to 3.0.6](https://github.com/microsoft/CCF/compare/ccf-3.0.2...ccf-3.0.6) which means a `t_cose` upgrade from [v1.1 to v1.1.1](https://github.com/laurencelundblade/t_cose/compare/v1.1...v1.1.1).
- [ISSUE-101](https://github.com/microsoft/scitt-ccf-ledger/issues/101): Replaced workaround code with `t_cose` for signature verification of Notary style claims.

Note: for testing, now that SCITT doesn't accept self-signed leaf certificates (which is what notation generates for testing), it's much easier to make use of X5ChainCertificateAuthority.create_chain() in test/infra/x5chain_certificate_authority.py to generate a cert chain with the correct extensions and flags, than to try to get the openssl incantation right.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,14 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+
+### Fixed
+- [ISSUE-101](https://github.com/microsoft/scitt-ccf-ledger/issues/101): Replaced workaround code with `t_cose` for signature verification of Notary style claims.
```
Comment (letmaik) [inline CHANGELOG.md:11]: Please stick to existing format of changelog entries. When `#xy` is copied to a release entry it is automatically linked.

File: app/src/cose.h [hunk 2]
```diff
@@ -691,7 +498,7 @@ namespace scitt::cose
     // auxiliary buffer size.
     t_cose_parameters params;
     t_cose_sign1_verify_init(
-      &verify_ctx, T_COSE_OPT_TAG_REQUIRED | T_COSE_OPT_DECODE_ONLY);
+      &verify_ctx, T_COSE_OPT_TAG_REQUIRED | T_COSE_OPT_DECODE_ONLY | T_COSE_OPT_UNKNOWN_CRIT_ALLOWED);
```
Comment (letmaik) [inline app/src/cose.h]: Given that this also affects non-Notary verification I think the `process_*_profile` functions in `verifier.h` should be extended to check that the crit field is empty for the other profiles. And there should be a comment here saying why using this flag is ok.

Comment (plietar) [reply app/src/cose.h]: Alternatively, add a flags argument to `cose::verify` that only the notary code path needs to set to `T_COSE_OPT_UNKNOWN_CRIT_ALLOWED`.

**Comment by plietar** [comment]
So https://github.com/microsoft/CCF/pull/5000 seems to have broken the unit tests.

We were using CCF's api to generate the certs used in unit tests, but they now set a path length constraint which breaks anything using intermediate CAs. I think the only workaround is for us to copy CCF's certificate generation code into `verifier_test.cpp` (or remove those tests).

**Comment by letmaik** [comment]
Can we do this in two steps? Updating of CCF should be a separate PR and we don't want to make that dependent on other work that's not strictly required for the update.

**Comment by darracott** [comment]
Superseded by https://github.com/microsoft/scitt-ccf-ledger/pull/132

**Pull request closed without merging**